cmd/result-aggregator/main.go

```diff
@@ -110,15 +116,43 @@ func withErrorRate(request *results.Request) {
 	errorRate.With(labels).Inc()
 }
 
-func handleCIOperatorResult(username string, password func() []byte) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		start := time.Now()
+type validator interface {
+	Validate(username, password string) bool
+}
+
+func newValidator(username string, password string, htpasswdFileRaw string) (validator, error) {
```

Comment:
Would be good if the validator re-read the file to be able to pick up changes to it without a restart

Reply:
Do we have an API for this already?

Reply:
an api? Just put the file reading in the verification closure

Reply:
Hmm. Each http request will need to read the file again?
I do not think we rotate that often.
Maybe just restarting the pod is a better choice?

Reply:
It will end up in the page cache if we do the request often. I don't think the performance is a big deal here. You could also create a goroutine that uses fsnotify to wach, but its not trivial, @AlexNPavel had some fun getting that to work properly for configmap/secret mounts

Reply:
Ack. Will update.

Reply:
Updated.